### PR TITLE
Javadoc

### PR DESCRIPTION
**Pull request by kortemik**
some javadoc

### CODE REVIEW

File: src/main/java/com/teragrep/rlp_03/channel/context/Context.java [hunk 1]
```diff
@@ -49,14 +49,31 @@
 import java.nio.channels.SelectionKey;
 import java.nio.channels.spi.AbstractSelectableChannel;
 
+/**
+ * Context is a network connection element, it may be one of the initiate, listen or established types.
```
Comment (StrongestNumber9) [inline src/main/java/com/teragrep/rlp_03/channel/context/Context.java:53]: Link  network connection element and types if possible

File: src/main/java/com/teragrep/rlp_03/channel/context/Context.java [hunk 2]
```diff
@@ -49,14 +49,31 @@
...
+ */
 public interface Context extends Closeable {
 
+    /**
+     * handles this context's event. Providing a non-related key will result in non-foreseen issues.
```
Comment (StrongestNumber9) [inline src/main/java/com/teragrep/rlp_03/channel/context/Context.java:58]: Start sentence with capital letter

File: src/main/java/com/teragrep/rlp_03/channel/context/Context.java [hunk 3]
```diff
@@ -49,14 +49,31 @@
...
     void handleEvent(SelectionKey selectionKey);
 
+    /**
+     * closes the underlying network connection element and frees resources attached to it.
+     */
     @Override
     void close(); // no exception is thrown
 
+    /**
+     * @return AbstractSelectableChannel of the network connection element.
```
Comment (StrongestNumber9) [inline src/main/java/com/teragrep/rlp_03/channel/context/Context.java:71]: Link the Channel

File: src/main/java/com/teragrep/rlp_03/channel/buffer/BufferLeasePool.java [hunk 4]
```diff
@@ -169,6 +182,9 @@ void internalOffer(BufferContainer bufferContainer) {
         }
     }
 
+    /**
+     * closes the pool, deallocating currently residing buffers and future ones when returned.
```
Comment (StrongestNumber9) [inline src/main/java/com/teragrep/rlp_03/channel/buffer/BufferLeasePool.java:186]: Maybe link the pool and buffers

File: src/main/java/com/teragrep/rlp_03/channel/context/ConnectContextFactory.java [hunk 5]
```diff
@@ -54,16 +54,31 @@
...
+/**
+ * Factory for creating {@link ConnectContext} type {@link Context}s.
+ */
 public class ConnectContextFactory {
 
     private final ExecutorService executorService;
     private final SocketFactory socketFactory;
 
+    /**
+     * @param executorService to handle connection's events with
```
Comment (StrongestNumber9) [inline src/main/java/com/teragrep/rlp_03/channel/context/ConnectContextFactory.java:66]: Link the ExecutorService and events

File: src/main/java/com/teragrep/rlp_03/channel/context/ConnectContextFactory.java [hunk 6]
```diff
@@ -54,16 +54,31 @@
...
         this.executorService = executorService;
         this.socketFactory = socketFactory;
     }
 
+    /**
+     * @param inetSocketAddress          address to initiate connection to.
+     * @param frameDelegate              for processing received data with.
+     * @param establishedContextConsumer for handling the callback once connection is established.
+     * @return ConnectContext to be registered with {@code {@link com.teragrep.rlp_03.EventLoop}.register()}.
+     * @throws IOException if underlying socketChannel is unable to initiate the connection.
```
Comment (StrongestNumber9) [inline src/main/java/com/teragrep/rlp_03/channel/context/ConnectContextFactory.java:80]: Link socketchannel

File: src/main/java/com/teragrep/rlp_03/channel/context/EstablishedContext.java [hunk 7]
```diff
@@ -48,11 +48,25 @@
 import com.teragrep.rlp_03.channel.InterestOps;
 import com.teragrep.rlp_03.channel.socket.Socket;
 
+/**
+ * Established type of Context. It produces ingress data into the provided
```
Comment (StrongestNumber9) [inline src/main/java/com/teragrep/rlp_03/channel/context/EstablishedContext.java:52]: Maybe link Context

File: src/main/java/com/teragrep/rlp_03/channel/context/RelpWrite.java [hunk 8]
```diff
@@ -51,9 +51,16 @@
...
 import java.util.function.Consumer;
 
+/**
+ * Egress frames are handled by this
+ */
 public interface RelpWrite extends Consumer<List<RelpFrameTX>>, Runnable {
 
-    // this must be thread-safe!
+    /**
+     * Sends asynchronously the frames provided. Implementation is required to be thread-safe.
```
Comment (StrongestNumber9) [inline src/main/java/com/teragrep/rlp_03/channel/context/RelpWrite.java:60]: Link the frame

File: src/main/java/com/teragrep/rlp_03/channel/context/RelpWrite.java [hunk 9]
```diff
@@ -51,9 +51,16 @@
 import java.util.concurrent.atomic.AtomicBoolean;
 import java.util.function.Consumer;
 
+/**
+ * Egress frames are handled by this
```
Comment (StrongestNumber9) [inline src/main/java/com/teragrep/rlp_03/channel/context/RelpWrite.java:55]: Link the frame

**Review by eemhu — COMMENTED** [review]
agree with previous comments

**Review by StrongestNumber9 — COMMENTED** [review]
Update looks good to me

File: src/main/java/com/teragrep/rlp_03/channel/socket/PlainSocket.java
Comment (eemhu) [inline src/main/java/com/teragrep/rlp_03/channel/socket/PlainSocket.java:1]: should Socket-related files also contain javadoc?

**Review by eemhu — APPROVED** [review]
otherwise, looks good to me

**Pull request merged**